src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -365,25 +365,167 @@ public class Options
+            {
+                byte[] buffer = new byte[_blockSize];
+                int offset = 0;
+                int totalLength = 0;
+                int bytesRead;
+
+                do
+                {
+                    bytesRead = await ReadAsync(buffer, offset, buffer.Length - offset);
+                    totalLength += bytesRead;
+                    offset+=bytesRead;
+
+                    if (bytesRead == buffer.Length)
+                    {
+                        byte[] newBuffer = new byte[buffer.Length + _blockSize];
+                        buffer.CopyTo(newBuffer, 0);
+                        offset = buffer.Length;
+                        buffer = newBuffer;
+                    }
+                } while (bytesRead > 0);
+
+                return System.Text.Encoding.ASCII.GetString(buffer, 0, totalLength);
+            }
+
+            public string ReadLine()
+            {
+                return ReadLineAsync().GetAwaiter().GetResult();
+            }
+
+            public async Task<string> ReadLineAsync()
+            {
+                int index = 0;
+                int startSearch = _start;
+
+                while (true)
+                {
+                    if (_start == _end || index == -1)
+                    {
+                        // We either have no data or we did not find LF in stream.
+                        // In either case, read more.
+                        if (_end + 2 > _buffer.Length)
+                        {
+                            // We no longer have space to read CRLF. Allocate new buffer and start over.
+                            byte[] newBuffer = new byte[_buffer.Length + _blockSize];
+                            int dataLength = _end - _start;
+                            if (dataLength > 0)
+                            {
+                                Array.Copy(_buffer, _start, newBuffer, 0, dataLength);
+                                _start = 0;
+                                _end = dataLength;
+                                _buffer = newBuffer;
+                            }
+                        }
+
+                        int bytesRead = await _stream.ReadAsync(_buffer, _end, _buffer.Length - _end).ConfigureAwait(false);
+                        if (bytesRead == 0)
+                        {
+                            break;
+                        }
+
+                        _end += bytesRead;
+                   }
+
+                    index = Array.IndexOf(_buffer, (byte)'\n', startSearch, _end - startSearch);
+                    if (index == -1)
+                    {
+                        // We did not find it, look for more data.
+                        startSearch = _end;
+                        continue;
+                    }
+
+                    int stringLength = index - _start;
+                    while (stringLength > 0 && _buffer[_start+stringLength] == '\n' || _buffer[_start+stringLength] == '\r')
```

Comment:
This seems too loose. Presumably you want to remove the CR, and also the LF if it's present. Can you just do that?

Reply:
updated.